### DISCUSSION TITLE
help with Bat script getting compression type of Tif files

### DISCUSSION THREAD

**Discussion by grfxlab** [ImageMagick/ImageMagick#7511] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

windows 11

### Description

I have a script to compress all tif files with LZW compression. To make it more efficient I want to run it on files not already compressed with LZW. I have tried to get the compression type using identify. The command works when running from a command prompt but it does not run correctly from within a bat file. I have a tif file named test.tif which has been compressed using lzw. When I run the following command 

identify -format "Compression: %[compression]" test.tif

It returns.... Compression: LZW

If I add the same command to a bat file it returns....
Compression: [compression]

As if it is not actually parsing the string.

I have tried no quotes, double quotes, single quotes, without luck. 

Any suggestions?

**Comment by GeeMack** [comment]
In a Windows  BAT script you need to double all the percent signs **except** the ones around Windows variables like %TEMP% etc.

    identify -format "Compression: %%[compression]" test.tif

**Comment by grfxlab** [comment]
GeeMack. Thank you. That worked perfectly.
